torch/testing/_internal/common_methods_invocations.py

```diff
@@ -896,7 +896,45 @@ def sample_inputs_diff(op_info, device, dtype, requires_grad, **kwargs):
 
     return tuple(sample_inputs)
 
-def sample_inputs_index_select(op_info, device, dtype, requires_grad, **kwargs):
+def sample_inputs_gradient(op_info, device, dtype, requires_grad):
+    test_cases = (
+        ((S,), None, None),
+        ((S,), 2.0, None),
+        ((S, S), None, None),
+        ((S, S), [2.0, 2.1], None),
+        ((S, S), [2.0, 2.1], (0, 1)),
+        ((4, 4, 4), [2.0, 1.0], (0, 1)),
+        ((3, 3, 3), [(1.0, 2.0, 3.0), (4.0, 5.0, 6.0), (7.0, 8.0, 9.0)], (0, 1, -1)),
+    )
+
+    sample_inputs = []
+    for size, spacing, dim in test_cases:
+        is_tensorlist = False
+        if isinstance(spacing, list):
+            is_tensorlist = True
+            for space in spacing:
+                if not isinstance(space, tuple):
+                    is_tensorlist = False
+                    break
+
+        t = make_tensor(size, device, dtype, low=None, high=None, requires_grad=requires_grad)
+        kwargs_dict = dict()
+        if dim is not None:
+            kwargs_dict['dim'] = dim
+        if is_tensorlist:
+            spacing_list = []
+            for space in spacing:
+                a = torch.tensor(space, dtype=torch.float16, device=device)
```

Comment:
Why dtype=torch.float16?

Reply:
Actually, let me change it to torch.float64. I believe spacing argument should always be interpreted as double.